### PR TITLE
UX improvements

### PR DESCRIPTION
**Pull request by mbakke**
Some minor UX improvements that annoyed me while monkeying around.  :)

### CODE REVIEW

File: zabbix_auto_config/processing.py [hunk 1]
```diff
@@ -225,7 +225,11 @@ def __init__(self, name, state, db_uri, host_modifier_dir):
     def get_host_modifiers(self):
         sys.path.append(self.host_modifier_dir)
 
-        module_names = [filename[:-3] for filename in os.listdir(self.host_modifier_dir) if filename.endswith(".py")]
+        try:
+            module_names = [filename[:-3] for filename in os.listdir(self.host_modifier_dir) if filename.endswith(".py")]
+        except(FileNotFoundError):
```
Comment (paalbra) [inline zabbix_auto_config/processing.py:230]: Parenthesis could be considered a linting error/warning. [C0325](https://pylint.pycqa.org/en/latest/messages/convention/superfluous-parens.html).

File: zabbix_auto_config/processing.py [hunk 2]
```diff
@@ -225,7 +225,11 @@ def __init__(self, name, state, db_uri, host_modifier_dir):
...
+            logging.error(f"Host modififier directory {host_modifier_dir} does not exist.")
```
Comment (paalbra) [inline zabbix_auto_config/processing.py:231]: Formatting should generally be passed as parameters. [W1203](https://pylint.pycqa.org/en/latest/messages/warning/logging-fstring-interpolation.html?highlight=W1203).

File: zabbix_auto_config/processing.py [hunk 3]
```diff
@@ -393,6 +397,8 @@ def disable_host(self, zabbix_host):
                 logging.info("Disabling host: '%s' (%s)", zabbix_host["host"], zabbix_host["hostid"])
             except pyzabbix.ZabbixAPIException as e:
                 logging.error("Error when disabling host '%s' (%s): %s", zabbix_host["host"], zabbix_host["hostid"], e.args)
+            except IndexError:
+                logging.error("Host group 'All-auto-disabled-hosts' not found, please create it.")
```
Comment (paalbra) [inline zabbix_auto_config/processing.py:401]: I feel that it is quite severe if the group is missing. Currently the application will crash, which I think is kind of "OK".

If you catch the exception nothing bad will happen other than a log event.

If it shouldn't crash maybe at least raise a `ZACException`?

And maybe keep logging the hostname that couldn't be disabled, like in the above except?

File: zabbix_auto_config/processing.py [hunk 4]
```diff
@@ -421,6 +427,8 @@ def enable_host(self, db_host):
                     logging.info("Enabling new host: '%s' (%s)", hostname, result["hostids"][0])
             except pyzabbix.ZabbixAPIException as e:
                 logging.error("Error when enabling/creating host '%s': %s", hostname, e.args)
+            except IndexError:
+                logging.error("Group 'All-hosts' does not exist, please create it.")
```
Comment (paalbra) [inline zabbix_auto_config/processing.py:431]: I guess my comment is the same as for disable_host.

**Review by paalbra — CHANGES_REQUESTED** [review]
More exception handling and logging is nice :+1:

I think however that the critical groups missing shouldn't be caught and only pass as a log event.

File: zabbix_auto_config/processing.py (hunk 2 above)
Comment (mbakke) [reply zabbix_auto_config/processing.py]: Thanks for this pointer, I had no idea :)

File: zabbix_auto_config/processing.py (hunk 3 above)
Comment (mbakke) [reply zabbix_auto_config/processing.py]: I dropped this commit for now, these groups are very important and should be handled in another way if at all.

File: zabbix_auto_config/processing.py [hunk 5]
```diff
@@ -225,7 +225,11 @@ def __init__(self, name, state, db_uri, host_modifier_dir):
     def get_host_modifiers(self):
         sys.path.append(self.host_modifier_dir)
 
-        module_names = [filename[:-3] for filename in os.listdir(self.host_modifier_dir) if filename.endswith(".py")]
+        try:
+            module_names = [filename[:-3] for filename in os.listdir(self.host_modifier_dir) if filename.endswith(".py")]
+        except FileNotFoundError:
+            logging.error("Host modififier directory %s does not exist.", host_modifier_dir)
```
Comment (paalbra) [inline zabbix_auto_config/processing.py:231]: Typo? Should be `self.host_modifier_dir`?

**Review by paalbra — CHANGES_REQUESTED** [review]

File: zabbix_auto_config/processing.py (hunk 5 above)
Comment (mbakke) [reply zabbix_auto_config/processing.py]: Good catch, thanks!

**Review by paalbra — APPROVED** [review]
:+1:

**Pull request merged**